internal/pkg/service/buffer/cleanup/task.go

```diff
@@ -0,0 +1,186 @@
+package cleanup
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/benbjohnson/clock"
+	etcd "go.etcd.io/etcd/client/v3"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/log"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/store/filestate"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/store/key"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/store/model"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/store/schema"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/store/slicestate"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/task"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/common/etcdop"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/common/etcdop/iterator"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/common/etcdop/op"
+	"github.com/keboola/keboola-as-code/internal/pkg/utils/errors"
+)
+
+type Task struct {
+	clock       clock.Clock
+	logger      log.Logger
+	schema      *schema.Schema
+	client      *etcd.Client
+	receiverKey key.ReceiverKey
+}
+
+func StartTask(k key.ReceiverKey) {
+}
+
+func newTask(d dependencies, logger log.Logger, k key.ReceiverKey) *Task {
+	return &Task{
+		clock:       d.Clock(),
+		logger:      logger,
+		schema:      d.Schema(),
+		client:      d.EtcdClient(),
+		receiverKey: k,
+	}
+}
+
+func (t *Task) Run(ctx context.Context) (task.Result, error) {
+	if err := t.cleanReceiver(ctx); err != nil {
+		return "", err
+	}
+	return fmt.Sprintf(`receiver "%s" has been cleaned`, t.receiverKey.String()), nil
+}
+
+func (t *Task) cleanReceiver(ctx context.Context) error {
+	errs := errors.NewMultiError()
+	if err := t.deleteExpiredTasks(ctx); err != nil {
+		errs.Append(err)
+	}
+	if err := t.deleteExpiredFiles(ctx); err != nil {
+		errs.Append(err)
+	}
+	return errs.ErrorOrNil()
+}
+
+// deleteExpiredTasks from the receiver.
+func (t *Task) deleteExpiredTasks(ctx context.Context) error {
+	deletedTasksCount := int64(0)
+
+	errs := errors.NewMultiError()
+	err := t.schema.
+		Tasks().
+		InReceiver(t.receiverKey).
+		GetAll().
+		Do(ctx, t.client).
+		ForEachKV(func(kv op.KeyValueT[model.Task], header *iterator.Header) error {
+			if kv.Value.IsForCleanup() {
+				if err := etcdop.Key(kv.Key()).Delete().DoOrErr(ctx, t.client); err == nil {
```

Comment:
Maže sa priamo dany etcd key, kedze sme tam mali nejake zmeny schemy.
Prefix per receiver sa nemenil, iba to co nasleduje dalej.